data/core/macros/items.cfg

```diff
@@ -261,11 +261,22 @@
 #enddef
 
 #define OBJ_POTION_HOLY X Y ID
-    {PICKUPPABLE_ITEM {ID} {X} {Y} side=1 items/holy-water.png
+    {PICKUPPABLE_ITEM {ID} {X} {Y} (
+        side=1
+        [has_attack]
+            range=melee
+        [/has_attack]
+        [not]
+            [has_attack]
+                range=melee
+                type=arcane
+            [/has_attack]
+        [/not]
+    ) items/holy-water.png
```

Comment:
Do you mean:

    [has_attack]
        range=melee
        [not]
            type=arcane
        [/not]
    [/has_attack]

I don't think so. That says "If I have any melee attack, one of which is not arcane."

What we want is "I have melee attacks, none of which are arcane."